ERCS/erc-7771.md

```diff
@@ -0,0 +1,647 @@
+        bytes4 sig4 = msg.sig;
+				address implementation;
+
+				// Yul lookup table: selector => implementation address
+				assembly {
+				    let sig32 := shr(224, sig4)
+
+				    function findImplementation(sig) -> result {
+				        if lt(sig,0x81d49b75) {
+				            if lt(sig,0x53a47bb7) {
+				                switch sig
+				                case 0x0f36a2de { result := _SYNTHS_MODULE }
+				                case 0x1627540c { result := _OWNER_MODULE }
+				                case 0x2d6b3a6b { result := _SYNTHS_MODULE }
+				                case 0x35eb2824 { result := _OWNER_MODULE }
+				                case 0x3659cfe6 { result := _UPGRADE_MODULE }
+				                case 0x51456061 { result := _SYNTHS_MODULE }
+				                leave
+				            }
+				            switch sig
+				            case 0x53a47bb7 { result := _OWNER_MODULE }
+				            case 0x57d7995a { result := _SYNTHS_MODULE }
+				            case 0x624bd96d { result := _OWNER_MODULE }
+				            case 0x7104fba5 { result := _SYNTHS_MODULE }
+				            case 0x718fe928 { result := _OWNER_MODULE }
+				            case 0x79ba5097 { result := _OWNER_MODULE }
+				        }
+				        if lt(sig,0xd1b6c504) {
+				            switch sig
+				            case 0x81d49b75 { result := _SNXTOKEN_MODULE }
+				            case 0x8da5cb5b { result := _OWNER_MODULE }
+				            case 0xa44896e7 { result := _SYNTHS_MODULE }
+				            case 0xaaf10f42 { result := _UPGRADE_MODULE }
+				            case 0xc6253216 { result := _SYNTHS_MODULE }
+				            case 0xc7f62cda { result := _UPGRADE_MODULE }
+				            leave
+				        }
+				        switch sig
+				        case 0xd1b6c504 { result := _SNXTOKEN_MODULE }
+				        case 0xdaac1198 { result := _SNXTOKEN_MODULE }
+				        case 0xe040da24 { result := _SNXTOKEN_MODULE }
+				        case 0xf4c4dc31 { result := _SYNTHS_MODULE }
+				        case 0xf5d6f068 { result := _SNXTOKEN_MODULE }
+				        case 0xfbc6aa6f { result := _SYNTHS_MODULE }
+				        leave
+				    }
+
+				    implementation := findImplementation(sig32)
+				}
+
+				if (implementation == address(0)) {
+				    revert UnknownSelector(sig4);
+				}
+
+        // Delegatecall to the implementation contract
+        assembly {
+            calldatacopy(0, 0, calldatasize())
+
+            let result := delegatecall(gas(), implementation, 0, calldatasize(), 0, 0)
+            returndatacopy(0, 0, returndatasize())
+
+            switch result
+            case 0 {
+                revert(0, returndatasize())
+            }
+            default {
+                return(0, returndatasize())
+            }
+        }
+    }
+}
+```
+
+### Router Code Generation
+
+Router code SHOULD NOT be written manually, and generated by automated tooling instead. Tooling MUST ensure unique function selectors across all modules and that there are no storage collisions between any of its modules.
```

Comment:
This paragraph is specifying implementation details. If a human wants to do this by hand, as long as they meet all the requirements, it should still be compliant with this standard.

PS: this is good content for the security considerations section.

PPS: don't use the uppercase keywords outside of the specification section.